projects/vdk-control-cli/src/vdk/internal/control/command_groups/job/deploy_cli_impl.py

```diff
@@ -31,14 +30,16 @@ class JobDeploy:
     ZIP_ARCHIVE_TYPE = "zip"
     ARCHIVE_SUFFIX = "-archive"
 
-    def __init__(self, rest_api_url: str, output):
+    def __init__(self, rest_api_url: str, output: str):
```

Comment:
This might be a discussion for another time, but I'm also not sure why the output format is injected through the constructor. I don't think components should be concerned with their output format. That should be configured externally somehow, if there is a convenient way.

Reply:
That's a good point. It would be better if the Printer is initialized outside and directly injected. Inversion of control/dependency injection and all that. 

There are some complications for some CLI commands (like `vdk deploy --update`) we have a custom message with instructions for users which is very specific for that command and is really needed for text output .

Ultimately we could apply the MVC pattern for CLI  (with this class being split into DeployCommandController and the actual printed output being done in DeployCommandView classes) but the code is not that complex in most commands currently to go in that direction.